File: tecnicell.client/src/app/Interfaces/business/Screen.ts
import { Branch } from "./Branch";
import { PhoneBrand } from "./PhoneBrand";
import { Sale } from "./Sale";

export interface Screen {
    screenCode:      string;
    brand:           string;
    name:            string;
    quantity:        number;
    width:           number;
    height:          number;
    salePrice:       number;
    warranty:        number;
    brandNavigation: PhoneBrand;
    screenHistories: ScreenHistory[];
}
export interface ScreenHistory {
    screenCode:         string;
    date:               Date;
    actionHistory:      string;
    toBranch:           string;
    description:        string;
    quantity:           number;
    saleCode:           string;
    saleCodeNavigation: Sale;
    toBranchNavigation: Branch;
}

File: Tecnicell.Server/Models/Entity/Screen.cs
using System;
using System.Collections.Generic;

namespace Tecnicell.Server.Models.Entity;

public partial class Screen
{
    public string ScreenCode { get; set; } = null!;

    public string? Brand { get; set; }

    public string? Name { get; set; }

    public decimal? Width { get; set; }

    public decimal? Height { get; set; }

    public decimal? SalePrice { get; set; }

    public string? ImageCode { get; set; }

    public int? Warranty { get; set; }

    public virtual Brand? BrandNavigation { get; set; }

    public virtual Image? ImageCodeNavigation { get; set; }

    public virtual ICollection<ScreenHistory> ScreenHistories { get; set; } = new List<ScreenHistory>();
}
